src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php

```diff
@@ -49,24 +49,53 @@ private function addDoctrineSection(ArrayNodeDefinition $rootNode)
                     ->info('If true, by default for string enumerations, generate DBAL types with an ENUM SQL declaration with enum values instead of a VARCHAR (Your platform must support it)')
                 ->end()
                 ->arrayNode('types')
+                    ->beforeNormalization()
+                        ->always(function ($values) {
+                            $legacyFormat = false;
+                            foreach ($values as $name => $config) {
+                                if (class_exists($name)) {
+                                    $legacyFormat = true;
+                                    // need to flip
+                                    break;
+                                }
+                            }
+                            $newValues = [];
+
+                            if ($legacyFormat) {
+                                foreach ($values as $name => $value) {
+                                    if (\is_string($value)) {
+                                        $newValues[$value] = $name;
+                                    } else {
+                                        $newValues[$value['name']] = $value + ['class' => $name];
+                                    }
+                                }
+                            } else {
+                                $newValues = $values;
+                            }
+
+                            return $newValues;
+                        })
+                    ->end()
                     ->validate()
                         ->ifTrue(static function (array $v): bool {return self::hasNonEnumKeys($v); })
                         ->then(static function (array $v) { self::throwsNonEnumKeysException($v); })
```

Comment:
```suggestion
                        ->ifTrue(static function (array $v): bool {return self::hasNonEnumClasses($v); })
                        ->then(static function (array $v) { self::throwsNonEnumClassesException($v); })
```

Reply:
Can't we add this validation on the `class` node directly now it's not the keys anymore?